interface/ceed-preconditioning.c

```diff
@@ -1961,6 +1961,105 @@ int CeedOperatorMultigridLevelCreateH1(CeedOperator op_fine, CeedVector p_mult_f
+    CeedOperatorField *input_fields_restrict;
+
+    // -- Check number suboperators
+    CeedCall(CeedOperatorGetNumSub(op_restrict, &num_sub_ops_restrict));
+    CeedCall(CeedOperatorGetSubList(op_restrict, &sub_ops_restrict));
+    if (num_sub_ops != num_sub_ops_restrict) return CeedError(ceed, CEED_ERROR_INCOMPATIBLE, "Restriction and Prolongation operators must match");
+
+    // -- Check scale vectors
+    CeedCall(CeedOperatorGetFields(sub_ops[0], NULL, &input_fields, NULL, NULL));
+    CeedCall(CeedOperatorGetFields(sub_ops_restrict[0], NULL, &input_fields_restrict, NULL, NULL));
+    CeedCall(CeedOperatorFieldGetVector(input_fields[1], &scale_prolong));
+    CeedCall(CeedOperatorFieldGetVector(input_fields_restrict[1], &scale_restrict));
+    if (scale_prolong != scale_restrict) return CeedError(ceed, CEED_ERROR_INCOMPATIBLE, "Restriction and Prolongation operators must match");
+  }
+
+  // Work vectors
+  CeedCall(CeedOperatorFieldGetElemRestriction(input_fields[1], &elem_restr));
+  CeedCall(CeedElemRestrictionCreateVector(elem_restr, &ones_l_vec, NULL));
+  CeedCall(CeedVectorSetValue(ones_l_vec, 1.0));
+  CeedCall(CeedElemRestrictionCreateVector(elem_restr, &mult_l_vec, NULL));
+  CeedCall(CeedVectorSetValue(mult_l_vec, 0.0));
+  CeedCall(CeedVectorGetLength(mult_l_vec, &mult_l_vec_len));
+  CeedCall(CeedVectorGetArray(mult_l_vec, CEED_MEM_HOST, &mult));
+
+  // Compute multiplicity across suboperators
+  for (CeedInt i = 0; i < num_sub_ops; i++) {
+    const CeedScalar *sub_mult;
+    CeedVector        sub_mult_l_vec, ones_e_vec;
+
+    CeedCall(CeedOperatorGetFields(sub_ops[i], NULL, &input_fields, NULL, NULL));
+    CeedCall(CeedOperatorFieldGetElemRestriction(input_fields[1], &elem_restr));
+    CeedCall(CeedElemRestrictionCreateVector(elem_restr, &sub_mult_l_vec, &ones_e_vec));
+    CeedCall(CeedVectorSetValue(sub_mult_l_vec, 0.0));
+    CeedCall(CeedElemRestrictionApply(elem_restr, CEED_NOTRANSPOSE, ones_l_vec, ones_e_vec, CEED_REQUEST_IMMEDIATE));
+    CeedCall(CeedElemRestrictionApply(elem_restr, CEED_TRANSPOSE, ones_e_vec, sub_mult_l_vec, CEED_REQUEST_IMMEDIATE));
+    CeedCall(CeedVectorGetArrayRead(sub_mult_l_vec, CEED_MEM_HOST, &sub_mult));
+    // -- Flag every node present in the current suboperator
+    for (CeedInt j = 0; j < mult_l_vec_len; j++)
+      if (sub_mult[j] > 0.0) mult[j] += 1.0;
+    CeedCall(CeedVectorRestoreArrayRead(sub_mult_l_vec, &sub_mult));
+    CeedCall(CeedVectorDestroy(&sub_mult_l_vec));
+    CeedCall(CeedVectorDestroy(&ones_e_vec));
+  }
+  CeedCall(CeedVectorRestoreArray(mult_l_vec, &mult));
+  CeedCall(CeedVectorReciprocal(mult_l_vec));
+
+  // Scale by multiplicity
+  for (CeedInt i = 0; i < num_sub_ops; i++) {
+    CeedVector scale_l_vec;
+
+    CeedCall(CeedOperatorGetFields(sub_ops[i], NULL, &input_fields, NULL, NULL));
+    CeedCall(CeedOperatorFieldGetVector(input_fields[1], &scale_l_vec));
+    CeedCall(CeedVectorPointwiseMult(scale_l_vec, scale_l_vec, mult_l_vec));
+  }
```

Comment:
If I'm reading correctly, this changes the behavior of each suboperator, which could be problematic when working with subdomain problems (e.g., we solve a coupled problem and also subdomain problems). We could check the reference count to confirm that we have exclusive access before mutating (just error unsupported if not) and consider using a nodal scaling vector later.